### PR TITLE
feat: Public Speakers Page: Add search box to search for a speaker name

### LINKED ISSUE #6753: Public Speakers Page: Add search box to search for a speaker name
Please add a search box "search speakers" on top of the public speakers list of events and search speaker names.

Screenshot from 2021-02-18 11-06-23

### PR DESCRIPTION
**Pull request by MeghalBisht**
Fixes #6753

#### Short description of what this resolves:

https://user-images.githubusercontent.com/61330148/109559250-6fbff680-7b00-11eb-891b-4650a6fe796e.mp4

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/routes/public/speakers.js [hunk 1]
```diff
@@ -30,25 +30,69 @@ export const SPEAKERS_FILTER = [
...
+                    op   : 'eq',
+                    val  : 'confirmed'
+                  }]
+                }
+              ]
+            }
+          }
+        ]
+      }
+    ];
```
Comment (iamareebjamal) [inline app/routes/public/speakers.js:74]: No need to redefine speaker filters, please delete

File: app/routes/public/speakers.js [hunk 2]
```diff
@@ -30,25 +30,69 @@ export const SPEAKERS_FILTER = [
...
-        infinityCache : 36000
+        filter       : filterOptions,
+        perPage      : 9,
+        startingPage : 1,
+        perPageParam : 'page[size]',
+        pageParam    : 'page[number]',
+        store        : eventDetails,
+        include      : 'sessions.track',
+        sort         : 'order',
+        public       : true
```
Comment (iamareebjamal) [inline app/routes/public/speakers.js:94]: Why removed cache?

File: app/routes/public/speakers.js (hunk 1 above)
Comment (MeghalBisht) [reply app/routes/public/speakers.js:74]: but then speakers are filtered in index page also.

Comment (iamareebjamal) [reply app/routes/public/speakers.js:74]: Use shallow cloing

```js
const filterOptions = {...SPEAKER_FILTERS};
```

File: app/routes/public/speakers.js (hunk 2 above)
Comment (MeghalBisht) [reply app/routes/public/speakers.js:94]: the filter only works on reload due to caching

**Comment by mariobehling** [comment]
Thanks, please also include position and company in the search. E.g. if I type in "Microsoft" I should get the result of speakers that have Microsoft as an organisation.

**Pull request merged**